src/api/worker/facades/UserFacade.ts

```diff
@@ -101,12 +118,69 @@ export class UserFacade implements AuthDataProvider {
 		return userGroupKey
 	}
 
-	getGroupKey(groupId: Id): Aes128Key {
-		return getFromMap(this.groupKeys, groupId, () => {
-			return decryptKey(this.getUserGroupKey(), this.getMembership(groupId).symEncGKey)
+	getGroupKey(groupId: Id): Versioned<AesKey> {
+		return getFromMap(this.currentGroupKeys, groupId, () => {
+			const groupMembership = this.getMembership(groupId)
+			return {
+				version: Number(groupMembership.groupKeyVersion),
+				object: decryptKey(this.getUserGroupKey().object, groupMembership.symEncGKey),
+			}
 		})
 	}
 
+	async loadSymGroupKey(groupId: Id, version: number, entityClient: EntityClient): Promise<AesKey> {
```

Comment:
I don't think loading is a good fit for UserFacade. It is meant to be just a data holder for the current data (which is also evident by the fact that you can't inject `EntityClient` in here). You probably want something separate that can work with keys.